extensions/github/src/helpers/repository.ts

```diff
@@ -34,15 +34,61 @@ export const WEB_IDES = [
   },
 ];
 
-import { LocalStorage } from "@raycast/api";
+import { execSync } from "node:child_process";
+import { existsSync } from "node:fs";
+import { homedir } from "node:os";
+
+import { LocalStorage, Toast, getPreferenceValues, showToast } from "@raycast/api";
 import { useCachedState } from "@raycast/utils";
 import { useEffect } from "react";
 
 import { ExtendedRepositoryFieldsFragment } from "../generated/graphql";
+import { getGitHubToken } from "../helpers/withGithubClient";
+
+import { getErrorMessage } from "./errors";
 
 const VISITED_REPOSITORIES_KEY = "VISITED_REPOSITORIES";
 const VISITED_REPOSITORIES_LENGTH = 25;
 
+export async function cloneAndOpen(repository: ExtendedRepositoryFieldsFragment) {
+  const { application, baseClonePath } = getPreferenceValues<Preferences>();
+  const applicationPath = application?.path || "/Applications/Visual Studio Code.app";
+  const clonePath = `${baseClonePath || "~/GitHub"}/${repository.nameWithOwner}`;
+  const openCommand = `open -a ${applicationPath.replaceAll(" ", "\\ ")} ${clonePath}`;
+
+  const toast = await showToast({
+    title: `Opening ${repository.nameWithOwner}`,
+    message: `at ${clonePath}`,
+    style: Toast.Style.Animated,
+  });
+
+  if (!existsSync(clonePath.replace("~", homedir()))) {
+    const gitHubToken = getGitHubToken();
+    const cloneUrl = `https://oauth:${gitHubToken}@github.com/${repository.nameWithOwner}`;
```

Comment:
I tried many commands, but in the end, I found this the most robust.

- SSH should be preferred but requires the user to set it up, which may lead to unpleasant failures. Also, when the user uses [1Password CLI](https://developer.1password.com/docs/ssh/) or other password manager integrations to protect their SSH keys, the SSH clone won't work because the user doesn't get the authentication prompt, and the command fails as unauthenticated.
- The `gh` CLI is an alternative method but forces the user to install the CLI. We could install it for them, but I don't like forcing such dependency when it's not needed and doesn't provide much additional value. In any case, it also requires more complex error-handling logic, making this command unnecessarily cumbersome. Finally, if the user protects the `gh` CLI token with password managers (e.g., [1Password](https://developer.1password.com/docs/cli/shell-plugins/github/)), it won't work for the same reason as SSH.

We already store and use a GitHub token, either user-generated or from OAuth, and it already has the permissions required for the clone, so it's pointless to ask the user for further authentications.